### ISSUE TITLE
Custom DRM/CDM and decoding method

### ISSUE THREAD

**Issue by AlvaroCeCe** [shaka-project/shaka-player#6081]
**Have you read the [Tutorials](http://shaka-player-demo.appspot.com/docs/api/tutorial-welcome.html)?** Yes

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?** Yes

**What version of Shaka Player are you using?**  3.0.10

**Please ask your question**

Hi all! 
I would like to use my own cipher for a video played in Shaka player. As far as I know, it's not possible to directly change/tweak Shaka's encrypter and I have to bypass the CDM (_content decryption module_) to use my custom implementation of a decoding mechanism. 

Basically, I'm trying to implement a custom DRM where I manage a key, use it to decode a video and watch it on Shaka player. If I don't have that key, I won't be able to watch that video.

It's my first time working with DRMs and Shaka player so I'm quite lost on this topic.
Right now, I'm trying to do now is to work with Shaka Network request/reply and catch the video there, cipher/decypher and then pass it to Shaka player. It's the only way I thought to bypass CDM and implement my cipher.

Thanks in advance.

**Comment by avelad** [comment]
What you would have to do is the following:
- Give a manifest without any type of encryption information (otherwise we would initialize the browser's DRM)
- Create response filters where you can capture the server's response and decrypt your content before the player can add it to MSE.

Note:
- This would not work in browsers that do not have MSE
- This does not add any type of security, since in the end the buffer is clear and there is no protection, it would be similar to HLS with AES-128
